xarray/tests/test_backends.py

```diff
@@ -166,7 +166,7 @@ def create_encoded_masked_and_scaled_data(dtype: np.dtype) -> Dataset:
 
 def create_unsigned_masked_scaled_data(dtype: np.dtype) -> Dataset:
     encoding = {
-        "_FillValue": np.int8(-1),
+        "_FillValue": 255,
```

Comment:
```suggestion
        "_FillValue": -1,
```

Reply:
This function seems to only be used to create a "decoded" representation of the data. In that case, the resulting `_FillValue` in `.encoding` will be `255` as the `.decode` pipeline will produce an unsigned `.attrs["_FillValue"]` that the mask code will use and will eventually be put in `.encoding` (not sure what part of the code does this transition but I don't think it matters for this discussion).

My change here was mostly to revert the test to the previous behavior in `main`.